pkg/controller/unreachable/unreachable_controller_test.go

```diff
@@ -0,0 +1,492 @@
+				if cond != nil {
+					t.Error("Did not expect to get the unreachable status condition.")
+				}
+			},
+		},
+		{
+			name: "unreachable condition should change from true to false",
+			existing: []runtime.Object{
+				getCDWithUnreachableConditionTrue(),
+				getKubeconfigSecret(),
+			},
+			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
+				cond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition)
+				if cond.Status != corev1.ConditionFalse {
+					t.Errorf("Did not get expected state for unreachable condition. Expected: \n%#v\nGot: \n%#v", corev1.ConditionFalse, cond.Status)
+				}
+			},
+		},
+		{
+			name: "With lastProbeTime, lastTransitionTime is current time-1hour unreachable condition should change from true to false",
+			existing: []runtime.Object{
+				func() *hivev1.ClusterDeployment {
+					lastProbeTime := metav1.NewTime(time.Now().Add(-(time.Hour * 1)))
+					lastTransitionTime := lastProbeTime
+					return getCDWithProbeAndTransitionTime(lastProbeTime, lastTransitionTime, corev1.ConditionTrue)
+				}(),
+				getKubeconfigSecret(),
+			},
+			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
+				cond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition)
+				if cond.Status != corev1.ConditionFalse {
+					t.Errorf("Did not get expected state for unreachable condition. Expected: \n%#v\nGot: \n%#v", corev1.ConditionFalse, cond.Status)
+				}
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fake.NewFakeClient(test.existing...)
+			rcd := &ReconcileRemoteMachineSet{
+				Client:                        fakeClient,
+				scheme:                        scheme.Scheme,
+				logger:                        log.WithField("controller", "unreachable"),
+				remoteClusterAPIClientBuilder: mockReachableClusterAPIClientBuilder,
+			}
+
+			namespacedName := types.NamespacedName{
+				Name:      testName,
+				Namespace: testNamespace,
+			}
+
+			_, err := rcd.Reconcile(reconcile.Request{NamespacedName: namespacedName})
+
+			if test.validate != nil {
+				cd := &hivev1.ClusterDeployment{}
+				err := fakeClient.Get(context.TODO(), namespacedName, cd)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+					return
+				}
+				test.validate(t, cd)
+			}
+
+			if err != nil && !test.expectError {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			if err == nil && test.expectError {
+				t.Errorf("Expected error but got none")
+			}
+		})
+	}
+}
+
+func TestIsTimeForConnectivityCheck(t *testing.T) {
+	tests := []struct {
+		name     string
+		cd       *hivev1.ClusterDeployment
+		expected bool
+		validate func(*testing.T, bool, bool)
+	}{
+		{
+			name: "test isTimeForConnectivityCheck() when lastProbeTime, lastTransitionTime is current time-1hour",
```

Comment:
These names become test names that you can run individually on the CLI, so long names and () can get a little clunky when copy pasting and such. I would suggest brevity here, maybe "lastProbeTime and lastTransitionTime under 1 hour. (test is redundant, as is isTimeForConnectivityCheck as our test function is named TestIsTimeForConnectivityCheck)

Reply:
(example from my history: go test ./pkg/controller/clusterdeployment/ -run TestClusterDeploymentReconcile/Ensure_managed_DNSZone_is_deleted_with_cluster_deployment)